employee-rostering-docs/src/main/asciidoc/ProjectStructure/ProjectStructure-section.adoc

```diff
@@ -0,0 +1,236 @@
+                                                   startDateTime, endDateTime))
+    then
+        scoreHolder.addSoftConstraintMatch(kcontext, -$rosterParametrization.getUndesiredTimeSlotWeight());
+end
+----
+
+TIP: The first line of the `when` clause is a technique to dynamically change the weight of the constraint. If `undesiredTimeSlotWeight` is 0 the constraint is disregarded.
+
+Condition: Given an employee's undesired date and time slot, there is a shift for this employee such that the date
+and time interval of the shift intersects the undesired date and time slot.
+
+Action: The soft score is decreased by _undesiredTimeSlotWeight_ units.
+
+*Desired time slot for an employee*
+
+[source,ruby]
+----
+rule "Desired time slot for an employee"
+    when
+        $rosterParametrization : RosterParametrization(desiredTimeSlotWeight != 0)
+        EmployeeAvailability(
+                state == EmployeeAvailabilityState.DESIRED,
+                $e : employee,
+                $startDateTime : startDateTime,
+                $endDateTime : endDateTime)
+        Shift(
+                employee == $e,
+                GwtJavaTimeWorkaroundUtil.doTimeslotsIntersect($startDateTime,$endDateTime,
+                                                   startDateTime, endDateTime))
+    then
+        scoreHolder.addSoftConstraintMatch(kcontext, +$rosterParametrization.getDesiredTimeSlotWeight());
+end
+----
+
+TIP: The first line of the `when` clause is a technique to dynamically change the weight of the constraint. If `desiredTimeSlotWeight` is 0 the constraint is disregarded.
+
+Condition:  Given an employee desired date and time slot, there is a shift for this employee such that the date and time interval of the shift intersects the desired date and time slot.
+
+Action: The soft score is increased by _desiredTimeSlotWeight_ units.
+
+*Employee is not rotation employee*
+
+[source,ruby]
+----
+rule "Employee is not rotation employee"
+    when
+    	$rosterParametrization : RosterParametrization(rotationEmployeeMatchWeight != 0)
+        Shift(
+                rotationEmployee != null, employee != null, employee != rotationEmployee)
+    then
+        scoreHolder.addSoftConstraintMatch(kcontext, -$rosterParametrization.getRotationEmployeeMatchWeight());
+end
+----
+
+TIP: The first line of the `when` clause is a technique to dynamically change the weight of the constraint. If `rotationEmployeeMatchWeight` is 0 the constraint is disregarded.
+
+IMPORTANT: In general, employees desire to work following a regular schedule: a rotation plan. This represents a
+starting point for the actual schedule that is influenced by other factors (e.g. temporary unavailability). For this
+reason, all Shifts are initialized with a `rotationEmployee`.
+
+Condition: There a shift that is assigned to an employee which is not the rotation employee.
+
+Action: The soft score is decreased by _rotationEmployeeMatchWeight_ units.
```

Comment:
We also need explanations for contracts, and GwtJavaTimeWorkaroundUtil do not exist any more